models/model_choice.py

```diff
@@ -32,6 +35,56 @@ def load_checkpoint(filename):
         raise FileNotFoundError(f"=> No model found at '{filename}'")
 
 
+def verify_weights(num_classes, weights):
+    """Verifies that the number of weights equals the number of classes if any are given
+    Args:
+        num_classes: number of classes defined in the configuration file
+        weights: weights defined in the configuration file
+    """
+    if num_classes == 1 and len(weights) == 2:
+        warnings.warn("got two class weights for single class defined in configuration file; will assume index 0 = background")
+    elif num_classes != len(weights):
+        raise ValueError('The number of class weights in the configuration file is different than the number of classes')
+
+
+def set_hyperparameters(params, num_classes, model, checkpoint, dontcare_val):
+    """
+    Function to set hyperparameters based on values provided in yaml config file.
+    Will also set model to GPU, if available.
```

Comment:
I don't see any part of this function dedicated to setting model to GPU, am I blind? If not, we should update the docstring. This statement may date from a while ago.